packages/client/src/workflow-client.ts

```diff
@@ -513,14 +523,53 @@ export class WorkflowClient extends BaseClient {
     workflowTypeOrFunc: string | T,
     options: WorkflowStartOptions<T>,
     interceptors: WorkflowClientInterceptor[]
-  ): Promise<string> {
+  ): Promise<WorkflowStartOutput> {
     const workflowType = extractWorkflowType(workflowTypeOrFunc);
     assertRequiredWorkflowOptions(options);
     const compiledOptions = compileWorkflowOptions(ensureArgs(options));
 
-    const start = composeInterceptors(interceptors, 'start', this._startWorkflowHandler.bind(this));
+    // Adapt legacy `start` interceptors to the new `startWithDetails` interface.
+    const adaptedInterceptors: WorkflowClientInterceptor[] = interceptors.map((i) => {
```

Comment:
Nits:
- Should probably move these adaptors to a distinct file. The `workflow-client.ts` file is already quite heavy.
- I think we should adapt interceptors on Client creation, so that we don't have to do this repeatedly here.